src/Idris/SetOptions.idr

```diff
@@ -43,13 +44,15 @@ candidateDirs dname pkg bounds
                             (stringToNatOrZ num :: toVersionNum rest)
 
     getVersion : String -> (String, PkgVersion)
-    getVersion str
-        = case span (/= '-') str of
-               (name, ver) => case strM ver of
-                                   StrNil => (name, MkPkgVersion [0])
-                                   StrCons _ ver =>
-                                      (name, MkPkgVersion (toVersionNum ver))
-               _ => (str, MkPkgVersion [0])
+    getVersion str =
+      -- split the dir name into parts concatenated by "-"
+      -- treating the last part as the version number
+      -- and the initial parts as the package name
+      case reverse $ split (== '-') str of
```

Comment:
Rather than reverse-reverse, I would introduce `unsnoc : List1 a -> (List a, a)`
that returns the beginning of a list & its last element. This will make the
intention clearer.

Reply:
Sounds reasonable. However, I need some guidance in the proper order of doing this. Eventually, `unsnoc` should go to `Data.List1` in base. But this will not be available to the compiler until the next minor release, right? So a copy of `unsnoc` should also go to `Libraries.Data.List.Extra`, probably with an annotation that it should be removed with the next release?

Reply:
`Libraries.Data.List1` but yeah that's the idea.

Reply:
You may need to `import Libraries.Data.List1 as Lib` and use `Lib.unsnoc`
to disambiguate between the two equal definitions.